models/rfcn.lua

```diff
@@ -0,0 +1,118 @@
+local nn = require 'nn'
+require 'cunn'
+require 'cudnn'
+
+local Convolution = cudnn.SpatialConvolution
+local Max = nn.SpatialMaxPooling
+local ReLU = cudnn.ReLU 
+local SBatchNorm = nn.SpatialBatchNormalization
+
+local function createModel(depth, multiFactor, preModel)
+   local function ShareGradInput(module, key)
+      assert(key)
+      module.__shareGradInputKey = key
+      return module
+   end
+   
+   -- add rcn after the feature maps
+   local function rcn(nInputPlane, nOutputPlane, multiFactor, w, h)
+      local s = nn.Sequential()
+      local sOutside = nn.Sequential()
+      s:add(SBatchNorm(nInputPlane))
+      s:add(ReLU(true))
+      -- reduce dimension
+      local nMiddle = nInputPlane / 2 
+      local conv0 = Convolution(nInputPlane, nMiddle, 1, 1)
+      -- initialize
+      conv0.weight:normal(0, 0.01)
+      conv0.bias:zero()
+      s:add(conv0)
+      s:add(ReLU(true))
+
+      local table = nn.ConcatTable()
+      local len3 = math.ceil(w / multiFactor)
+      local len4 = math.ceil(h / multiFactor)
+      -- build the n-part layer
+      for i = 0, multiFactor - 1 do
+         for j = 0, multiFactor - 1 do
+            local part = nn.Sequential()
+            local offset3 = math.floor(i * w / multiFactor) + 1
+            if offset3 + len3 > w + 1 then
+               offset3 = w - len3 + 1
+            end
+            local offset4 = math.floor(j * h / multiFactor) + 1
+            if offset4 + len4 > h + 1 then
+               offset4 = w - len4 + 1
+            end
+            part:add(nn.Narrow(3, offset3, len3))
+            part:add(nn.Narrow(4, offset4, len4))
+            -- 1x1 Convolution
+            local conv1 = Convolution(nMiddle, nOutputPlane, 1, 1)
+            -- initialize
+            conv1.weight:normal(0, 0.01)
+            conv1.bias:zero()
+            part:add(conv1)
+            part:add(Max(len3, len4, 1, 1))
+            table:add(part)
+         end
+      end
+
+      s:add(table)
+      s:add(nn.CAddTable())
+      -- Balance the RFCN output and fc-layer output
+      local mul = nn.Mul()
```

Comment:
There seems to be an nn.MulConstant module for this purpose. But it's fine to keep the current implementation.